android/src/main/java/com/pagopa/ioreactnativecrypto/X509VerificationUtil.kt

```diff
@@ -0,0 +1,300 @@
+      for (cert in certificateChain) {
+        for (crl in crls) {
+          // Only check against CRLs issued by the certificate's issuer
+          if (cert.issuerX500Principal == crl.issuerX500Principal) {
+            val revokedCert = crl.getRevokedCertificate(cert)
+            if (revokedCert != null) {
+              return CertificateValidationResult(
+                isValid = false,
+                validationStatus = CertificateValidationStatus.REVOKED,
+                errorMessage = "Certificate revoked: ${cert.subjectX500Principal}, revocation date: ${revokedCert.revocationDate}"
+              )
+            }
+          }
+        }
+      }
+
+      // Create a CertPath from the certificate chain.
+      val certPath = certificateFactory.generateCertPath(certificateChain)
+
+      // Set up PKIX parameters using the trust anchor.
+      val pkixParams = PKIXParameters(setOf(trustAnchor)).also {
+        it.isRevocationEnabled = crls.isNotEmpty()
+      }
+
+      // Add CRLs to CertStore if any were downloaded
+      if (crls.isNotEmpty()) {
+        val certStore = CertStore.getInstance(
+          "Collection",
+          CollectionCertStoreParameters(crls)
+        )
+        pkixParams.addCertStore(certStore)
+      }
+
+      // Validate the certificate chain.
+      val validator = CertPathValidator.getInstance("PKIX")
+      validator.validate(certPath, pkixParams)
+
+      return CertificateValidationResult(
+        isValid = true,
+        validationStatus = CertificateValidationStatus.VALID
+      )
+    } catch (e: CertPathValidatorException) {
+      return CertificateValidationResult(
+        isValid = false,
+        validationStatus = CertificateValidationStatus.INVALID_CHAIN,
+        errorMessage = "Certificate path validation failed: ${e.message}"
+      )
+    } catch (e: VerificationException) {
+      return when (e) {
+        VerificationException.CRLExpired ->
+          CertificateValidationResult(
+            isValid = false,
+            validationStatus = CertificateValidationStatus.CRL_EXPIRED,
+            errorMessage = "CRL not defined"
+          )
+
+        VerificationException.CRLNotDefined ->
+          CertificateValidationResult(
+            isValid = false,
+            validationStatus = CertificateValidationStatus.CRL_NOT_DEFINED,
+            errorMessage = "CRL expired"
+          )
+      }
+
+    } catch (e: Exception) {
+      return CertificateValidationResult(
+        isValid = false,
+        validationStatus = CertificateValidationStatus.VALIDATION_ERROR,
+        errorMessage = "Validation error: ${e.message}"
+      )
+    }
+  }
+
+  /**
+   * Public method to verify certificate chains with synchronous API
+   */
+  fun verifyCertificateChain(
+    certChainBase64: List<String>,
+    trustAnchorCertBase64: String
+  ): CertificateValidationResult {
+    return runBlocking {
+      verifyCertificateChainAsync(certChainBase64, trustAnchorCertBase64)
+    }
```

Comment:
Is `runBlocking` a good choice here? Why not using a background thread like you do in the iOS counterpart? What do you think?